test/integration_tests.jl

```diff
@@ -0,0 +1,155 @@
+# You should set up a Git repo with an example Julia package for the integration tests.
+# You should create a bot user GitHub account, and give the bot user GitHub account push access to the testing repo.
+# You should generate a personal access token (PAT) for the bot user GitHub account.
+# You should add the PAT as an encrypted Travis CI environment variable named `BCBI_TEST_USER_GITHUB_TOKEN`.
+# You should generate an SSH deploy key. Upload the public key as a deploy key for the testing repo.
+# You should make the private key an encrypted Travis CI environment variable named `COMPATHELPER_PRIV`. Probably you will want to Base64-encode it.
+
+@testset "GitHub Integration Test" begin
+    GLOBAL_PR_TITLE_PREFIX = Random.randstring(8)
+    COMPATHELPER_INTEGRATION_TEST_REPO = ENV["COMPATHELPER_INTEGRATION_TEST_REPO"]
+    TEST_USER_GITHUB_TOKEN = ENV["BCBI_TEST_USER_GITHUB_TOKEN"]
+
+    api = GitHub.GitHubAPI(; token=GitHub.Token(TEST_USER_GITHUB_TOKEN))
+    user, _ = GitForge.get_user(api)
+    repo, _ = GitForge.get_repo(api, COMPATHELPER_INTEGRATION_TEST_REPO)
+
+    url = "https://$(user.login):$(TEST_USER_GITHUB_TOKEN)@github.com/$(COMPATHELPER_INTEGRATION_TEST_REPO)"
+
+    # Setup variables used in CompatHelper.main()
+    env = Dict(
+        "GITHUB_REPOSITORY" => COMPATHELPER_INTEGRATION_TEST_REPO,
+        "GITHUB_TOKEN" => TEST_USER_GITHUB_TOKEN,
+    )
+    ci_cfg = CompatHelper.GitHubActions(
+        user.login,
+        "41898282+github-actions[bot]@users.noreply.github.com"
+    )
+
+    @testset "master_1" begin
```

Comment:
can we maybe get some comments on what each `master_?` test is specifically testing for?

Reply:
I added in a field `description` in each of the respective `Project.toml` files.